### ISSUE TITLE
Create circle around coordinates

### ISSUE THREAD

**Issue by buu-huu** [Mapsui/Mapsui#1150]
Is there any possibility to get a circle around a point with the actual diameter? I have the latitude, longitude and the diameter of a Mesocyclone object. Currently I use a default value for the radius while drawing a circle using a Polygon. Is there any possibility, to draw the circle around that point with the actual size (e.g. 10km)?
My current code:

```csharp
        private static Layer CreateMesoDiameterLayer(List<Mesocyclone> mesocyclones)
        {
            var features = new Features();
            return new Layer("Diameter Layer")
            {
                DataSource = new MemoryProvider(CreateDiameterCircles(mesocyclones)),
                Style = new VectorStyle
                {
                    Fill = new Brush(new Color(255, 97, 247, 50)),
                    Outline = new Pen
                    {
                        Color = new Color(121, 97, 255, 200),
                        Width = 2,
                        PenStyle = PenStyle.LongDashDot,
                        PenStrokeCap = PenStrokeCap.Butt
                    }
                }
            };
        }

        private static List<Polygon> CreateDiameterCircles(List<Mesocyclone> mesocyclones)
        {
            var result = new List<Polygon>();

            foreach (var meso in mesocyclones)
            {
                var polygon = new Polygon();
                var middle = FromLongLat(meso.Longitude, meso.Latitude);
                double radius = 100000; // Currently default value. Here I want to have the actual diameter (property of a mesocyclone

                for (double angle = 0; angle <= 2 * Math.PI; angle += 0.2)
                {
                    polygon.ExteriorRing.Vertices.Add(
                        new Point(
                            middle.X + radius * Math.Cos(angle),
                            middle.Y + radius * Math.Sin(angle)));
                }

                result.Add(polygon);
            }

            return result;
        }
```

**Comment by buu-huu** [comment]
I dont't know, if theres a better way, but I managed to do the job with a little bit of maths. Maybe, it will help someone in the future, so I share my solution with you.

Basically I created a function, that creates a new Point from a given `longitude`, `latitude`, `distance` and `bearing/rotation`:

```csharp
private const double RADIUS_EQUATOR = 6378.1;

private static Point GetDistancePoint(double longitude, double latitude, double rotation, double distance)
{
    double brng = ConvertDegreesToRadians(rotation); // In radians! 90° == 1.57

    double lat1 = ConvertDegreesToRadians(latitude);
    double lon1 = ConvertDegreesToRadians(longitude);

    double lat2 = Math.Asin(Math.Sin(lat1) * Math.Cos(distance / RADIUS_EQUATOR) +
        Math.Cos(lat1) * Math.Sin(distance / RADIUS_EQUATOR) * Math.Cos(brng));

    double lon2 = lon1 + Math.Atan2(Math.Sin(brng) * Math.Sin(distance / RADIUS_EQUATOR) * Math.Cos(lat1),
        Math.Cos(distance / RADIUS_EQUATOR) - Math.Sin(lat1) * Math.Sin(lat2));

    lat2 = ConvertRadiansToDegrees(lat2);
    lon2 = ConvertRadiansToDegrees(lon2);
    return FromLongLat(lon2, lat2);
}
```

When I create the layer for the diameter rings, I just call that function in a for loop (<= 360 degrees) and add the generated point to the polygon:

```csharp
private static List<Polygon> CreateDiameterCircles(List<Mesocyclone> mesocyclones)
{
    var result = new List<Polygon>();

    foreach (var meso in mesocyclones)
    {
        var polygon = new Polygon();
        var mesoCenterLon = meso.Longitude;
        var mesoCenterLat = meso.Latitude;
        double radius = meso.Diameter/2;

        for (int step = 0; step <= 360; step += 5)
        {
            polygon.ExteriorRing.Vertices.Add(
                GetDistancePoint(mesoCenterLon, mesoCenterLat, step, radius));
        }
        result.Add(polygon);
    }
    return result;
}

/* Math Helpers */

private static Point FromLongLat(double longitude, double latitude)
{
    return SphericalMercator.FromLonLat(longitude, latitude);
}

public static double ConvertRadiansToDegrees(double radians)
{
    double degrees = (180 / Math.PI) * radians;
    return degrees;
}

public static double ConvertDegreesToRadians(double degrees)
{
    double radians = (Math.PI / 180) * degrees;
    return radians;
}
```

**Comment by janusw** [comment]
Well, you don't need to do all of this manually. There is the class `Mapsui.UI.Forms.Circle` just for this purpose:

```csharp
                var middle = FromLongLat(meso.Longitude, meso.Latitude);
                double radius = 10000;

                var c = new Circle
                {
                    Center = middle,
                    Radius = Distance.FromMeters(radius),
                    StrokeWidth = 1,
                    Quality = 60
                };
```

Make sure to use a large enough `Quality` parameter, so that you get an actual circle and not some edgy polygon ;)

**Comment by buu-huu** [comment]
Thanks Janus, actually I can't include the class `Circle` to my project :/ Are there any steps to do before, I can use `Circle`? I use WPF for my project, maybe there are some conflicts?

**Comment by charlenni** [comment]
I assume, that you don't use Xamarin.Forms. That's the reason why you don't have this class. But you could look, how the Xamarin.Forms implementation does it.

**Comment by janusw** [comment]
> Thanks Janus, actually I can't include the class `Circle` to my project :/ Are there any steps to do before, I can use `Circle`? 

Referencing the Mapsui.Forms package and `using Mapsui.UI.Forms` should be enough. I recommend using the existing class instead of re-implementing it.

Note that the `Circle` class still has bugs, though (e.g. #1136, which hopefully will be fixed soon, see #1138).
